aptos-move/framework/supra-framework/sources/multisig_voting.move

```diff
@@ -0,0 +1,1327 @@
+    /// this proposal.
+    /// @param min_vote_threshold The minimum number of votes needed to consider this proposal successful.
+    /// @param expiration_secs The time in seconds at which the proposal expires and can potentially be resolved.
+    /// @param metadata A simple_map that stores information about this proposal.
+    /// @return The proposal id.
+    public fun create_proposal<ProposalType: store>(
+        proposer: address,
+        voting_forum_address: address,
+        execution_content: ProposalType,
+        execution_hash: vector<u8>,
+        min_vote_threshold: u64,
+        voters: vector<address>,
+        expiration_secs: u64,
+        metadata: SimpleMap<String, vector<u8>>,
+    ): u64 acquires VotingForum {
+        create_proposal_v2(
+            proposer,
+            voting_forum_address,
+            execution_content,
+            execution_hash,
+            min_vote_threshold,
+            voters,
+            expiration_secs,
+            metadata,
+            false
+        )
+    }
+
+    /// Create a single-step or a multi-step proposal with the given parameters
+    ///
+    /// @param voting_forum_address The forum's address where the proposal will be stored.
+    /// @param execution_content The execution content that will be given back at resolution time. This can contain
+    /// data such as a capability resource used to scope the execution.
+    /// @param execution_hash The sha-256 hash for the execution script module. Only the same exact script module can
+    /// resolve this proposal.
+    /// @param min_vote_threshold The minimum number of votes needed to consider this proposal successful.
+    /// @param voters The list of voters who are eligible to vote.".
+    /// @param expiration_secs The time in seconds at which the proposal expires and can potentially be resolved.
+    /// @param metadata A simple_map that stores information about this proposal.
+    /// @param is_multi_step_proposal A bool value that indicates if the proposal is single-step or multi-step.
+    /// @return The proposal id.
+    public fun create_proposal_v2<ProposalType: store>(
+        proposer: address,
+        voting_forum_address: address,
+        execution_content: ProposalType,
+        execution_hash: vector<u8>,
+        min_vote_threshold: u64,
+        voters: vector<address>,
+        expiration_secs: u64,
+        metadata: SimpleMap<String, vector<u8>>,
+        is_multi_step_proposal: bool,
+    ): u64 acquires VotingForum {
+
+        // Make sure the execution script's hash is not empty.
+        assert!(vector::length(&execution_hash) > 0, error::invalid_argument(EPROPOSAL_EMPTY_EXECUTION_HASH));
+
+        // Make sure voters length must grether than or equal to vote threshold
+        assert!(vector::length(&voters) >= min_vote_threshold, error::invalid_argument(ETHRESHOLD_EXCEEDS_VOTERS));
```

Comment:
Also, assertion is required that `min_vote_threshold>1` (must require more than one approval)